File: Leetcode/Daily/07-2025/C++/17.cpp
// Problem: 3202. Find the Maximum Length of Valid Subsequence II (Medium)
// Link: https://leetcode.com/problems/find-the-maximum-length-of-valid-subsequence-ii

class Solution {
public:
    int maximumLength(vector<int>& nums, int k) {
        int ans = 0;
        int n = nums.size();
        
        for (int i = 0; i < k; i++) {
            vector<int> v(k);
            for (int j = 0; j < n; j++) {
                v[nums[j] % k] = v[((i - nums[j]) % k + k ) % k] + 1;
                ans = max(ans, v[nums[j] % k]);
            }
        }

        return ans;
    }
};

File: Leetcode/Daily/07-2025/Python3/17.py
# Problem: 3202. Find the Maximum Length of Valid Subsequence II (Medium)
# Link: https://leetcode.com/problems/find-the-maximum-length-of-valid-subsequence-ii

class Solution:
    def maximumLength(self, nums: List[int], k: int) -> int:
        ans = 0
        n = len(nums)

        for i in range(k):
            v = [0] * k
            for j in range(n):
                v[nums[j] % k] = v[((i - nums[j]) % k + k ) % k] + 1
                ans = max(ans, v[nums[j] % k])

        return ans;         